docker/types/containers.py

```diff
@@ -433,6 +434,30 @@ def __init__(self, version, binds=None, port_bindings=None,
                 raise host_config_version_error('volume_driver', '1.21')
             self['VolumeDriver'] = volume_driver
 
+        if cpu_count:
+            if not isinstance(cpu_count, int):
+                raise host_config_type_error('cpu_count', cpu_count, 'int')
+            if version_lt(version, '1.25'):
+                raise host_config_version_error('cpu_count', '1.25')
+
+            self['CpuCount'] = cpu_count
+
+        if cpu_percent:
+            if not isinstance(cpu_percent, int):
+                raise host_config_type_error('cpu_percent', cpu_percent, 'int')
+            if version_lt(version, '1.25'):
+                raise host_config_version_error('cpu_percent', '1.25')
+
+            self['CpuPercent'] = cpu_percent
+
+        if cpus:
+            if not isinstance(cpus, (float, int)):
+                raise host_config_type_error('cpus', cpus, 'float')
+            if version_lt(version, '1.25'):
+                raise host_config_version_error('cpus', '1.25')
+
+            self['NanoCpus'] = int(1000000000 * cpus)
```

Comment:
Let's have `nano_cpus` be the parameter instead, limit the acceptable types to `int` and avoid `float` uncertainty altogether.